services/discourse_push_notifications/pusher.rb

```diff
@@ -64,9 +55,33 @@ def self.clear_subscriptions(user)
       user.custom_fields[DiscoursePushNotifications::PLUGIN_NAME] = {}
     end
 
-    def self.subscribe(user, subscription)
+    def self.subscribe(user, subscription, send_confirmation)
       subscriptions(user)[extract_unique_id(subscription)] = subscription.to_json
       user.save_custom_fields(true)
+      if send_confirmation == "true"
+        message = {
+          title: I18n.t("discourse_push_notifications.popup.confirm_title",
+                        site_title: SiteSetting.title),
+          body: I18n.t("discourse_push_notifications.popup.confirm_body"),
+          icon: SiteSetting.logo_small_url || SiteSetting.logo_url,
+          tag: "#{Discourse.current_hostname}-subscription"
+        }
+
+        begin
+          response = Webpush.payload_send(
```

Comment:
I think we should extract the common code in `Pusher` into a method for this method to call.